io.openems.edge.goodwe.et/src/io/openems/edge/goodwe/et/ess/GoodWeEtBatteryInverterImpl.java

```diff
@@ -482,7 +474,9 @@ private void updatechannels() {
 		 */
 		final Channel<Integer> batteryPower = this.channel(EssChannelId.P_BATTERY1);
 		Integer activePower = batteryPower.getNextValue().get();
+		Integer productionPower = null;
 		for (AbstractGoodWeEtCharger charger : this.chargers) {
+			productionPower = TypeUtils.sum(productionPower, charger.getActualPower().get());
 			activePower = TypeUtils.sum(activePower, charger.getActualPowerChannel().getNextValue().get());
```

Comment:
Can you explain why you use `getNextValue` for one and `value` for the other one?

According to the logic: wouldn't it be sufficient to just sum up the `productionPower` and later add that value to `activePower`?